tools/build-utils.js

```diff
@@ -1,248 +1,48 @@
+			copyOverDataJSON(`${file}/${f}`);
+		} else if (f.endsWith('.json')) {
+			fs.copyFileSync(`${file}/${f}`, require('path').resolve('dist', `${file}/${f}`));
 		}
-	});
-}
-
-function copyOverDataJSON(file) {
-	const source = './data/' + file;
-	const dest = './.data-dist/' + file;
-	fs.readFile(source, function (err, text) {
-		if (err) throw err;
-		fs.writeFile(dest, text, function (err) {
-			if (err) throw err;
-		});
-	});
-}
-
-exports.transpile = (doForce, decl) => {
-	if (doForce) force = true;
-	if (sucrase('./config', './.config-dist')) {
-		replace('.config-dist', [
-			{regex: /(require\(.*?)(lib|sim)/g, replace: `$1.$2-dist`},
-		]);
-	}
-
-	if (sucrase('./data', './.data-dist')) {
-		replace('.data-dist', [
-			{regex: /(require\(.*?)(lib|sim)/g, replace: `$1.$2-dist`},
-		]);
-	}
-
-	if (sucrase('./sim', './.sim-dist')) {
-		replace('.sim-dist', [
-			{regex: /(require\(.*?)\/(lib|data|config)/g, replace: `$1/.$2-dist`},
-			{regex: /(from '.*?)\/(lib|data|config)/g, replace: `$1/.$2-dist`},
-		]);
-	}
-
-	sucrase('./lib', './.lib-dist');
-
-	if (sucrase('./server', './.server-dist')) {
-		replace('.server-dist', [
-			{regex: /(require\(.*?)(data|lib|sim)/g, replace: `$1.$2-dist`},
-		]);
-	}
-
-	sucrase('./translations', './.translations-dist');
-
-	if (sucrase('./tools', './tools', null, ['.', 'sets', 'simulate'])) {
-		replace('tools', [
-			{regex: /(require\(.*?)(lib|sim|server)/g, replace: `$1.$2-dist`},
-		]);
 	}
+};
 
-	if (!fs.existsSync('./.data-dist/README.md')) {
-		const text = '**NOTE**: This folder contains the compiled output of the `data/` directory.\n' +
-			'You should be editing the `.ts` files there and then running `npm run build` or\n' +
-			'`./pokemon-showdown` to force these `.js` files to be recreated.\n';
-		fs.writeFile('./.config-dist/README.md', text.replace('data/', 'config/'), function () {});
-		fs.writeFile('./.data-dist/README.md', text, function () {});
-		fs.writeFile('./.sim-dist/README.md', text.replace('data/', 'sim/'), function () {});
-		fs.writeFile('./.server-dist/README.md', text.replace('data/', 'server/'), function () {});
-		fs.writeFile('./.translations-dist/README.md', text.replace('data/', 'translations/'), function () {});
-		fs.writeFile('./.lib-dist/README.md', text.replace('data/', 'lib/'), function () {});
+exports.transpile = (decl) => {
+	shell(
+		'git ls-files "*.ts" "*.tsx" | grep -v global |' +
+		'xargs node_modules/.bin/esbuild --log-level=error --outbase=. --outdir=./dist ' +
+		`--format=cjs --tsconfig=./tsconfig.json`
+	);
+
+	const gitignored = ['./server/chat-plugins/private'];
+	for (const path of gitignored) {
+		if (fs.existsSync(path)) {
+			shell([
+				`find ${path}`, `grep "\\.[tj]s\\b"`, `grep -vF ".d.ts"`, `grep -v "node_modules"`, `grep -v global`,
+				`xargs node_modules/.bin/esbuild --log-level=error --outbase=${path} --outdir=./dist/${path} ` +
+				`--format=cjs --tsconfig=./tsconfig.json`,
+			].join(' | '), true);
```

Comment:
Has this been tested on Windows?

Reply:
Yes, see above - Wob tested all of this on Windows with me.